core/site.go

```diff
@@ -82,8 +95,74 @@ func NewSiteFromConfig(
 	site.tariffs = tariffs
 	site.savings = NewSavings(tariffs)
 
+	// maxCurrent is set and > 0: expect load management to be applied
+	site.gridLimiter = *NewLimiter("grid")
+	if site.MaxGridCurrent > 0.0 {
+		site.gridLimiter.SetMargin(0)
+		site.gridLimiter.Enable(true)
+	} else if site.MaxGridCurrent < 0.0 {
+		// no support for negative values
+		return nil, errors.New("maxCurrent for the site must not be negative")
+	} else { // 0.0, default when not configured, or set to 0
+		site.gridLimiter.SetTotalLimit(0.0)
+		site.gridLimiter.Enable(false)
+	}
+
+	// give all loadpoints the gridLimiter, and also tell the gridLimiter its members
+	for _, curLp := range site.loadpoints {
+		site.gridLimiter.loadpoints = append(site.gridLimiter.loadpoints, curLp)
+		curLp.limiters = append(curLp.limiters, &site.gridLimiter)
+	}
+	// now apply loadpoints to circuits
+	var ccNames []string // for checking LP have correct cc name
+	for _, curCC := range site.Circuits {
+		lm := NewLimiter(curCC.Name)
+		lm.Enable(true)
+		lm.SetTotalLimit(curCC.MaxCurrent)
+		ccNames = append(ccNames, curCC.Name)
+		// iterate LPs to assign to matching circuit
+		for _, curLp := range site.loadpoints {
+			if curLp.Circuit == curCC.Name {
```

Comment:
Gehört sowas nicht in den Loadpoint statt in die Site?

Reply:
Im Moment wo die LP von der Konfig angelegt werden, gibts die Site mit den Circuits noch nicht. Wenn die Site angelegt wird, dann bekommt sie von der Konfig die LP mitgegeben. Daher hier auch die Initialisierung.